CMakeLists.txt

```diff
@@ -37,58 +44,58 @@ if(NOT WIN32)
   endif()
 endif()
 
+add_definitions(-Wall -Wextra -std=gnu1x -Wno-unused-parameter)
+if (CMAKE_COMPILER_IS_GNUCC)
+  if (WITH_COVERAGE)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fprofile-arcs -ftest-coverage")
+    set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -fprofile-arcs -ftest-coverage")
+  endif ()
+endif ()
+
+set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -O0")
+
 check_symbol_exists(ffsl "string.h" HAS_FFSL)
 if(NOT HAS_FFSL)
-	add_definitions(-DNO_FFSL)
+  add_definitions(-DNO_FFSL)
 endif()
 
 #-------------------------------------
-# Verify PostgreSQL Version
+# Verify PostgreSQL version
 #-------------------------------------
+
 set(PG_MIN_MAJOR_VERSION "10")
 set(PG_MAX_MAJOR_VERSION "14")
 
-#--------------------------------
 find_package(POSTGRESQL ${PG_MIN_MAJOR_VERSION} REQUIRED)
 if (NOT POSTGRES_VERSION VERSION_LESS PG_MAX_MAJOR_VERSION)
   message(FATAL_ERROR "Not supporting PostgreSQL ${POSTGRESQL_VERSION}")
 endif()
 
-add_definitions(-DPOSTGRESQL_VERSION_NUMBER=${POSTGRESQL_VERSION_NUMBER} -DPOSTGRESQL_VERSION_STRING="${POSTGRESQL_VERSION_STRING}")
+add_definitions(-DPOSTGRESQL_VERSION_NUMBER=${POSTGRESQL_VERSION_NUMBER}
+  -DPOSTGRESQL_VERSION_STRING="${POSTGRESQL_VERSION_STRING}")
 
 include_directories(SYSTEM ${POSTGRESQL_INCLUDE_DIR})
 if(WIN32)
-    include_directories(SYSTEM ${POSTGRESQL_INCLUDE_DIR}/port/win32)
-    if(MSVC)
-        include_directories(SYSTEM ${POSTGRESQL_INCLUDE_DIR}/port/win32_msvc/)
-    endif()
+  include_directories(SYSTEM ${POSTGRESQL_INCLUDE_DIR}/port/win32)
+  if(MSVC)
+    include_directories(SYSTEM ${POSTGRESQL_INCLUDE_DIR}/port/win32_msvc/)
+  endif()
 endif()
 
 if(WIN32)
-    link_directories(${POSTGRESQL_LIBRARIES})
-    link_libraries(postgres)
+  link_directories(${POSTGRESQL_LIBRARIES})
+  link_libraries(postgres)
 endif()
 
+#-------------------------------------
+# Verify PostGIS version
+#-------------------------------------
+
 # Currently does not work with other versions of PostGIS
 find_package(POSTGIS 2.5 EXACT REQUIRED)
 add_definitions(-DPOSTGIS_VERSION_STR="${POSTGIS_VERSION_STR}")
 add_definitions(-DPOSTGIS_VERSION_NUMBER=${POSTGIS_VERSION_NUMBER})
 
-#--------------------------------
-
-
-add_definitions(-Wall -Wextra -std=gnu1x -Wno-unused-parameter)
```

Comment:
Why remove these checks?
Remember a warning in one system might be an error in another

Reply:
See comment below